buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesGraphTask.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.DependencySet;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A task to generate a dependency graph of our runtime dependencies and push that via
+ * an API call to a given endpoint of a SCA tool/service.
+ * The graph is built according to the specification in https://github.com/snyk/dep-graph#depgraphdata
+ *
+ * Due to the nature of our dependency resolution in gradle, we are abusing the aforementioned graph definition as
+ * the graph we construct has a single root ( the subproject ) and all dependencies are children of that root,
+ * irrespective of if they are direct dependencies or transitive ones ( that should be children of other children ).
+ * Although we end up lacking some contextual information, this allows us to scan and monitor only the dependencies
+ * that are bundled and used in runtime.
+ */
+public class DependenciesGraphTask extends DefaultTask {
+
+    private Configuration runtimeConfiguration;
+    private Configuration compileOnlyConfiguration;
```

Comment:
Why are we including compile-only dependencies? These aren't bundled with Elasticsearch so I'm not sure we should be including them in our report. Security vulnerabilities, by definition, should only be applicable to runtime code.

Reply:
We are not including them, we are resolving them here so that we can exclude them. See https://github.com/elastic/elasticsearch/pull/63641/files#diff-09a991b3b2c89b70a65e2e26cccdb4ee8c24ce440f82706e44aa79948dad51c1R119

Reply:
The resolving runtime dependencies process is picked up from our dependenciesInfo task